src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+            Environment_DrawSandstorm(this, this->envCtx.sandstormState);
+        }
+
+        if (this->unk_18876 != 0) {
+            Environment_FillScreen(gfxCtx, 0, 0, 0, this->unk_18876, 3);
+        }
+
+        DebugDisplay_DrawObjects(this);
+        Play_DrawMotionBlur(this);
+
+        if (((R_PAUSE_MENU_MODE == 1) || (gTrnsnUnkState == 1)) || (R_PICTOGRAPH_PHOTO_STATE == 1)) {
+            Gfx* sp74;
+            Gfx* sp70 = POLY_OPA_DISP;
+
+            sp74 = Graph_GfxPlusOne(sp70);
+            gSPDisplayList(OVERLAY_DISP++, sp74);
+            this->pauseBgPreRender.fbuf = gfxCtx->curFrameBuffer;
+
+            if (R_PAUSE_MENU_MODE == 1) {
+                R_PAUSE_MENU_MODE = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = this->unk_18E58;
+            } else if (R_PICTOGRAPH_PHOTO_STATE == 1) {
+                R_PICTOGRAPH_PHOTO_STATE = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = this->unk_18E58;
+            } else {
+                gTrnsnUnkState = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = NULL;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167DE4.s")
+            func_801705B4(&this->pauseBgPreRender, &sp74);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167F0C.s")
+            if (this->pauseBgPreRender.cvgSave != NULL) {
+                func_80170730(&this->pauseBgPreRender, &sp74);
+            }
 
+            gSPEndDisplayList(sp74++);
+            Graph_BranchDlist(sp70, sp74);
+            POLY_OPA_DISP = sp74;
+            this->unk_18B49 = 2;
+            SREG(33) |= 1;
+        } else {
+        Play_Draw_PostWorldDraw:
+            Play_PostWorldDraw(this);
+        }
+    }
+
+    //! Fake: regalloc around here
+    if (this && this && this) {}
+
+    if ((this->view.unk164 != 0) && !gDbgCamEnabled) {
+        Vec3s sp4C;
+
+        Camera_Update(&sp4C, GET_ACTIVE_CAM(this));
+        View_UpdateViewingMatrix(&this->view);
+        this->view.unk164 = 0;
+        if ((this->skyboxId != 0) && !this->envCtx.skyboxDisabled) {
+            SkyboxDraw_UpdateMatrix(&this->skyboxCtx, this->view.eye.x, this->view.eye.y, this->view.eye.z);
+        }
+    }
+
+    if (!sp25B) {
+        func_800FE3E0(this);
+    }
+
+    CLOSE_DISPS(this->state.gfxCtx);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Play_Draw.s")
+#endif
+
+void Play_SetupDraw(PlayState* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+
+    {
+        GraphicsContext* gfxCtx2 = this->state.gfxCtx;
+
+        if (sBombersNotebookOpen) {
+            if (D_801FBBD4 != 1) {
+                MsgEvent_SendNullTask();
+                func_80178818();
+                gfxCtx2->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx2->framebufferIdx % 2);
```

Comment:
```suggestion
                gfxCtx2->curFrameBuffer = SysCfb_GetFbPtr(gfxCtx2->framebufferIndex % 2);
```
Since it's in this function?